### PR TITLE
*: Support for changing column

### PR DESCRIPTION
**Pull request by zimulala**
Support for changing column name and changing column type. The behavior of changing column type like modifying type.
see [url](http://dev.mysql.com/doc/refman/5.7/en/alter-table.html)

### CODE REVIEW

**Comment by zimulala** [comment]
PTAL @hanfei1991 @tiancaiamao @coocood @shenli

**Comment by zimulala** [comment]
PTAL @coocood

File: ddl/ddl.go [hunk 1]
```diff
@@ -1103,40 +1107,81 @@ func (d *ddl) modifiable(origin *types.FieldType, to *types.FieldType) bool {
...
+	if len(spec.NewColumn.Name.Table.O) != 0 && ident.Name.L != spec.NewColumn.Name.Table.L {
+		return errWrongTableName.GenByArgs(spec.NewColumn.Name.Table.O)
+	}
+	if len(spec.OldColumnName.Table.O) != 0 && ident.Name.L != spec.OldColumnName.Table.L {
+		return errWrongTableName.GenByArgs(spec.OldColumnName.Table.O)
+	}
+
+	newColName := spec.NewColumn.Name
+	spec.NewColumn.Name = spec.OldColumnName
+	job, err := d.getModifiableColumnJob(ctx, ident, spec)
```
Comment (coocood) [inline ddl/ddl.go]: This is too tricky, add a `columnName` argument to `getModifiableColumnJob`, so we don't need to set `NewColumn.Name` to `OldColumnName` and later update job Args.

File: parser/parser_test.go [hunk 2]
```diff
@@ -288,6 +288,7 @@ func (s *testParserSuite) TestDMLStmt(c *C) {
 		{"ALTER TABLE t DISABLE KEYS", true},
 		{"ALTER TABLE t ENABLE KEYS", true},
 		{"ALTER TABLE t MODIFY COLUMN a varchar(255)", true},
+		{"ALTER TABLE t CHANGE COLUMN a b varchar(255)", true},
```
Comment (shenli) [inline parser/parser_test.go]: Please also add test case here: https://github.com/pingcap/tidb/blob/master/parser/parser_test.go#L41

**Comment by shenli** [comment]
Rest LGTM

**Comment by coocood** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

**Pull request merged**